### ISSUE TITLE
[Bug]: Qwen3-VL-Embedding model produces different embeddings than official qwen_vl_utils implementation

### ISSUE THREAD

**Issue by namburiamit** [vllm-project/vllm#33204]
### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Your output of `python collect_env.py` here
```

</details>

## Summary

When using `Qwen/Qwen3-VL-Embedding-8B` with vLLM's pooling runner, the generated embeddings differ significantly from the official implementation using `qwen_vl_utils`, resulting in ~0.92 cosine similarity instead of 1.0 for identical video inputs.

## Environment

- **vLLM version:** v0.14.0
- **Model:** `Qwen/Qwen3-VL-Embedding-8B`
- **Hardware:** NVIDIA A100 80GB
- **OS:** Ubuntu 22.04

### vLLM Server Configuration
```bash
python3 -m vllm.entrypoints.openai.api_server \
    --port 8080 \
    --model Qwen/Qwen3-VL-Embedding-8B \
    --served-model-name Qwen3VL-Embedding-8B \
    --runner pooling \
    --tensor-parallel-size 1 \
    --gpu-memory-utilization 0.90 \
    --trust-remote-code \
    --max-model-len 8192 \
    --mm-processor-kwargs '{"fps": 2.0, "max_frames": 64}' \
    --pooler-config '{"pooling_type": "LAST", "normalize": true}'
```

### Official Implementation (qwen_vl_utils)
```python
from qwen_vl_utils import process_vision_info
from transformers import Qwen2VLForConditionalGeneration, AutoProcessor

model = Qwen2VLForConditionalGeneration.from_pretrained(
    "Qwen/Qwen3-VL-Embedding-8B",
    torch_dtype=torch.bfloat16
)
processor = AutoProcessor.from_pretrained(
    "Qwen/Qwen3-VL-Embedding-8B",
    min_pixels=256*28*28,
    max_pixels=1280*28*28
)

messages = [
    {"role": "system", "content": "You are a helpful assistant."},
    {"role": "user", "content": [
        {"type": "video", "video": video_path, "fps": 2.0, "max_frames": 64},
        {"type": "text", "text": instruction}
    ]}
]

# Key difference: uses qwen_vl_utils for video processing
image_inputs, video_inputs, video_kwargs = process_vision_info(
    messages, return_video_kwargs=True
)

inputs = processor(
    text=[text],
    images=image_inputs,
    videos=video_inputs,
    padding=True,
    return_tensors="pt",
    **video_kwargs
)

# Get last token embedding
with torch.no_grad():
    outputs = model(**inputs, output_hidden_states=True)
    embeddings = outputs.hidden_states[-1][:, -1, :]
    embeddings = F.normalize(embeddings, p=2, dim=-1)
```

## Problem

The embeddings from vLLM and the official implementation have only **~0.92 cosine similarity** for the same video input, even when all configurable parameters match:

| Parameter | vLLM | Official |
|-----------|------|----------|
| fps | 2.0 | 2.0 |
| max_frames | 64 | 64 |
| pooling | LAST token | LAST token |
| normalize | true | true |

### Impact on Retrieval Tasks

In my video retrieval evaluation, I noticed a significant difference in recall:

| Implementation | Recall@1 |
|---------------|----------|
| Official (qwen_vl_utils) | **~80%** |
| vLLM v0.14.0 | **~55%** |

This is a significant degradation for production use cases.

## Root Cause Analysis

The difference appears to stem from **video frame extraction and preprocessing**:

1. **Official implementation** uses `qwen_vl_utils.process_vision_info()` which handles video decoding with specific logic for frame sampling and pixel preprocessing

2. **vLLM** uses its internal `Qwen2VLProcessor` which processes videos differently

Even with identical `fps` and `max_frames` settings, the actual pixel values extracted differ between the two pipelines.

## Reproduction

```python
import numpy as np
from scipy.spatial.distance import cosine

# Get embedding from vLLM
vllm_embedding = get_vllm_embedding(video_path)

# Get embedding from official implementation  
official_embedding = get_official_embedding(video_path)

# Compare
similarity = 1 - cosine(vllm_embedding, official_embedding)
print(f"Cosine similarity: {similarity}")  # ~0.92, expected 1.0
```

## Requested Solution

This would resolve this:

1. Integrate `qwen_vl_utils.process_vision_info()` as the video processor for Qwen3-VL models to match official behavior

## References

- Qwen3-VL-Embedding model: https://huggingface.co/Qwen/Qwen3-VL-Embedding-8B
- qwen_vl_utils: https://github.com/QwenLM/Qwen2.5-VL (contains process_vision_info)

## Additional Context

This issue affects production deployments where users have pre-computed embeddings using the official implementation and want to serve the model via vLLM without re-indexing their entire corpus.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by noooop** [comment]
cc @DarkLight1337 @Isotr0py

**Comment by DarkLight1337** [comment]
vLLM doesn't use `qwen_vl_utils` because it's outside of the HF `AutoProcessor` and is an optional step (otherwise Transformers would also include `qwen_vl_utils` logic if it's really that important). I suggest you use `qwen_vl_utils` to preprocess the images/videos manually before sending them to OpenAI server.

**Comment by tzyingwu** [comment]
I’ve encountered the same issue described here and wanted to add some additional context along with a workaround that has been effective on my side.
From my observations, the root cause appears to be how vllm serve currently handles the preprocessing logic required by `qwen_vl_utils`, particularly for multimodal inputs. As a result, directly serving these models through the built-in server path can lead to discrepancies compared to expected behavior.  

As a workaround, I’ve been able to get things working reliably by:
- Using the **vLLM offline engine** instead of `vllm serve`
- Wrapping the preprocessing step from `qwen_vl_utils` and **vLLM offline engine** inside a **LitServe** endpoint
- Feeding the properly preprocessed inputs into `vLLM` for generation

With this setup, I’m seeing cosine similarity compared to transformers of:
- ~0.92 for image inputs
- ~0.97 for video inputs

Overall, this approach ensures the expected multimodal preprocessing is correctly applied and avoids the limitations of the current serve path. While this is a workaround rather than a proper fix, it may be useful for others encountering the same issue until native support or a more integrated solution becomes available.
Happy to share more details if helpful.

**Comment by DarkLight1337** [comment]
It's intended that `qwen_vl_utils` is outside the scope of vLLM as that's also how it's done in Transformers and we want to keep consistent with it. Maybe we should mention this explicitly in Supported Models page... @noooop do you have time to do that?
